docs/asciidoc/static/complex-pipeline.asciidoc

```diff
@@ -0,0 +1,363 @@
+
+[float]
+[[twitter-configuration]]
+==== Reading from a Twitter feed
+
+To add a Twitter feed, you need several pieces of information:
+
+* A _consumer_ key, which uniquely identifies your Twitter app, which is Logstash in this case.
+* A _consumer secret_, which serves as the password for your Twitter app.
+* One or more _keywords_ to search in the incoming feed.
+* An _oauth token_, which identifies the Twitter account using this app.
+* An _oauth token secret_, which serves as the password of the Twitter account.
+
+Visit https://dev.twitter.com/apps to set up a Twitter account and generate your consumer key and secret, as well as 
+your OAuth token and secret.
+
+Use this information to add the following lines to the input section of the `first-pipeline.conf` file:
+
+[source,json]
+twitter {
+    consumer_key =>
+    consumer_secret =>
+    keywords =>
+    oauth_token =>
+    oauth_token_secret => 
+}
+
+[float]
+[[configuring-lsf]]
+==== The Logstash Forwarder
+The Logstash Forwarder is a software client that processes data at the machine of origin before sending the data to the 
+Logstash instance. Data from the Logstash Forwarder uses a secure protocol called _lumberjack_ that is designed for 
+reliability and low latency. Default Logstash configuration includes the lumberjack input plugin, which is designed to 
+be resource-friendly. To install the Logstash Forwarder on your data source machine, install the appropriate package 
+from the main Logstash https://www.elastic.co/downloads/logstash[product page].
+
+Create a configuration file for the Logstash Forwarder similar to the following example:
+
+[source,json]
+{
+  "network": {
+    "servers": [ "localhost:5043" ],
+
+    "ssl ca": "/path/to/localhost.crt",
+
+    "timeout": 15
+  },
+
+  "files": [
+    {
+      "paths": [
+        "/path/to/sample-log"
+      ],
+      "fields": { "type": "apache" }
+    }
+  ]
+}
+
+Save this configuration file as `logstash-forwarder.conf`. 
+
+Configure your Logstash instance to use the Lumberjack input plugin by adding the following lines to the `input` section 
+of the `first-pipeline.conf` file:
+
+[source,json]
+lumberjack {
+    port => “5043”
+    ssl_certificate => “/path/to/ssl-cert”
+    ssl_key => “/path/to/ssl-key”
+}
+
+[float]
+[[logstash-file-output]]
+==== Writing Logstash Data to a File
+
+You can configure your Logstash pipeline to write data directly to a file with the 
+{logstash}plugins-outputs-file.html[`file`] output plugin.
+
+Configure your Logstash instance to use the `file` output plugin by adding the following lines to the `output` section 
+of the `first-pipeline.conf` file:
+
+[source,json]
+file {
+    path => /path/to/target/file
+}
+
+[float]
+[[multiple-es-nodes]]
+==== Writing to multiple Elasticsearch nodes
+
+Writing to multiple Elasticsearch nodes lightens the resource demands on a given Elasticsearch node, as well as 
```

Comment:
Does this distribute the writes across the nodes, or write to every specified node? I presume from this first sentence it's the former--if so it would be good to state that explicitly.